Source/bmalloc/libpas/src/libpas/pas_try_allocate_common.h

```diff
@@ -165,9 +167,11 @@ pas_try_allocate_common_impl_slow(
             
             pas_physical_memory_transaction_begin(&transaction);
             pas_heap_lock_lock();
-            
-            result = pas_large_heap_try_allocate(
-                &heap->large_heap, size, alignment, config.config_ptr, &transaction);
+                        
+            if (pas_probabilistic_guard_malloc_should_call_pgm())
+                result = pas_large_heap_try_allocate_pgm(&heap->large_heap, size, alignment, config.config_ptr, &transaction);
+            else
+                result = pas_large_heap_try_allocate(&heap->large_heap, size, alignment, config.config_ptr, &transaction);
```

Comment:
This is not OK to place a function call in this path. We should compute enable_pgm at the initialization of global heap once, and read global variable here.

Reply:
That's a good point. I'll look into updating the design.